### PR TITLE
feat: introduce DefaultExtensionCatalog

### PR DESCRIPTION
**Pull request by vbarua**
contains static strings for default extensions

### CODE REVIEW

**Comment by vbarua** [comment]
It's come up a number of times that we're copying the names of extensions in various places. This change adds a `DefaultExtensionCatalog` class whose sole purpose is to hold static strings of extension names for re-use.

File: core/src/main/java/io/substrait/type/YamlRead.java [hunk 1]
```diff
@@ -25,7 +26,8 @@ public class YamlRead {
 
   public static void main(String[] args) throws Exception {
     try {
-      System.out.println("Read: " + YamlRead.class.getResource("/functions_boolean.yaml"));
+      System.out.println(
```
Comment (vibhatha) [inline core/src/main/java/io/substrait/type/YamlRead.java:29]: nit: @vbarua just curious do we need a println here? Or should we have used logs?

**Review by vibhatha — APPROVED** [review]
This PR LGTM!

File: core/src/main/java/io/substrait/type/YamlRead.java (hunk 1 above)
Comment (vbarua) [reply core/src/main/java/io/substrait/type/YamlRead.java:29]: This looks to be some old debugging code. Trying to run the main function throws a `FileNotFoundException` for me, so I'm not sure if it works anymore. I could be worth removing, but I think that would be better as it's own change.

**Pull request merged**